src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -77,6 +195,21 @@ GcsTaskManager::GcsTaskManagerStorage::AddOrReplaceTaskEvent(
   int32_t attempt_number = events_by_task.attempt_number();
   TaskAttempt task_attempt = std::make_pair<>(task_id, attempt_number);
 
+  // Update the parent <-> children index if parent info available.
+  // This could be done first before merging/adding to the storage because this is
+  // independent of the current events or the to-be-added index position.
+  // NOTE: it's possible the parent_task_id is not in the storage/index (due to eviction
+  // or parent task event not reported yet.)
+  {
+    TaskID parent_task_id =
+        events_by_task.has_task_info()
+            ? TaskID::FromBinary(events_by_task.task_info().parent_task_id())
```

Comment:
is parent id reported from every event? Or when we first submit a task?

Reply:
Only when we have `task_info` included.

Reply:
Is task_info included in all events? Is there no concern that this parent_task_id is the default value because we didn't update this?

Reply:
`task_info` is not included in all events - only some of them (the first status update). 
We treat default value (Nil) as indication of no parent, so we also check against if `parent_task_id` is Nil.

Reply:
that makes sense. Can you write a comment (task_info is only included in the first event) to this code block?